src/hidimstat/conditional_sampling.py

```diff
@@ -0,0 +1,180 @@
+            when `data_type` is "auto".
+
+        """
+        self.data_type = data_type
+        self.model_regression = model_regression
+        self.model_binary = model_binary
+        self.model_categorical = model_categorical
+
+        if data_type == "auto":
+            self.model_auto = {
+                "continuous": model_regression,
+                "binary": model_binary,
+                "categorical": model_categorical,
+            }
+        elif data_type == "continuous":
+            self.model = model_regression
+        elif data_type == "binary":
+            self.model = model_binary
+        elif data_type == "categorical":
+            self.model = model_categorical
+        else:
+            raise ValueError(f"type of data '{data_type}' unknow.")
+        self.rng = check_random_state(random_state)
+        self.categorical_max_cardinality = categorical_max_cardinality
+
+    def fit(self, X: np.ndarray, y: np.ndarray):
+        """
+        Fit the model that estimates $\mathbb{E}[y | X]$.
+
+        Parameters
+        ----------
+        X : ndarray
+            The variables used to predict the group of variables $y$.
+        y : ndarray
+            The group of variables to predict.
+        """
+
+        if self.data_type == "auto":
+            if len(np.unique(y)) == 2:
+                self.data_type = "binary"
+                self.model = self.model_auto["binary"]
+            elif len(np.unique(y)) <= self.categorical_max_cardinality:
+                self.data_type = "categorical"
+                self.model = self.model_auto["categorical"]
+            else:
+                self.data_type = "continuous"
+                self.model = self.model_auto["continuous"]
+
+        # Group of variables
+        if (y.ndim > 1) and (y.shape[1] > 1):
+            if self.data_type in ["binary", "categorical"]:
+                self.model = MultiOutputClassifier(self.model)
+            elif self.data_type == "continuous" and not isinstance(
+                self.model, NATIVELY_MULTIOUTPUT
+            ):
+                self.model = MultiOutputRegressor(self.model)
+            self.multioutput_ = True
+        else:
+            self.multioutput_ = False
+        self.model.fit(X, y)
+
+    def sample(self, X: np.ndarray, y: np.ndarray, n_samples: int = 1) -> np.ndarray:
+        """
+        Sample from the conditional distribution $p(X^j | X^{-j})$.
+
+        Parameters
+        ----------
+        X : ndarray
+            The complementary of the considered set of variables, $X^{-j}$.
+        y : ndarray
+            The group of variables to sample, $X^j$.
+        n_samples : int, optional
+            The number of samples to draw.
+
+        Returns
+        -------
+        ndarray
+            An array of shape (n_samples, y.shape[1]) containing the samples.
+        """
+
+        check_is_fitted(self.model)
+        if (self.data_type in ["binary", "categorical"]) and (
```

Comment:
Why is an error for binary and categorical types of variable and not for continuous types of variables?

I think this error should be moved to the `elif` after. This avoids checking twice the same things.

Reply:
- For continuous data, sampling from the conditional is approximated by shuffling the residual from the predictions. For binary & categorical, it is done by sampling from the predicted probability of the classes. For methods that don't return predicted probabilities, it is not obvious how to sample from the conditional $p(X^j|X^{-j})$. 
 -  agreed, I moved it

Reply:
You can add a test for knowing if the model as a `predict` method. It's not worse to add it.